mapper/src/main/java/org/jboss/pnc/mapper/BuildBCRevisionFetcher.java

```diff
@@ -48,10 +66,20 @@ public void mockBrewAttributes(BuildRecord build, @MappingTarget Build.Builder d
         Integer id = build.getBuildConfigurationId();
         Integer revision = build.getBuildConfigurationRev();
 
-        BuildConfigurationAudited buildConfigurationAudited = bcAuditedRepository
-                .queryById(new IdRev(id, revision));
+        // If somebody before us already set the BCA we don't need to query it from DB again
+        BuildConfigurationAudited bca = build.getBuildConfigurationAudited();
+        if(bca == null){
+            bca = bcAuditedRepository.queryById(new IdRev(id, revision));
+        }
```

Comment:
@vibe13 I added the null check to skip the DB query if it is already set. I don't think that current implementation retrieves the BCA sometimes before this, but it can be useful if you find a place that can be optimized e.g. by querying a bunch of BCAs in one query or some other optimization.